src/ray/core_worker/core_worker.cc

```diff
@@ -103,6 +103,18 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
   gcs_client_ = std::make_shared<gcs::RedisGcsClient>(gcs_options);
   RAY_CHECK_OK(gcs_client_->Connect(io_service_));
 
+  // Subscribe to all unexpected worker failure notifications from the local
+  // and remote raylets. Note that this does not include workers that failed
+  // due to node failure.
+  const auto &worker_failed = [this](gcs::RedisGcsClient *client, const WorkerID &id,
+                                     const gcs::WorkerFailureData &worker_failure) {
+    HandleWorkerFailed(id, worker_failure);
+  };
+  RAY_CHECK_OK(gcs_client_->worker_failure_table().Subscribe(
+      JobID::Nil(), ClientID::Nil(), worker_failed,
+      /*subscribe_callback=*/nullptr,
+      /*done_callback=*/nullptr));
```

Comment:
Right now all the gcs tables are hidden inside the gcs_client accessor interface [gcs_client.h](https://github.com/ray-project/ray/blob/master/src/ray/gcs/gcs_client.h). Could we add an accessor also for this instead of using the table directly?

Reply:
I'm not really sure about this because I don't understand the accessors well enough, but if you have an idea of which accessor this should go under, let me know. You're also welcome to do that after this PR is merged.

Reply:
Basically you could add a `WorkerInfoAccessor` like existing gcs accessors in [accessor.h](https://github.com/ray-project/ray/blob/master/src/ray/gcs/accessor.h), for more details you could search for `ActorInfoAccessor` for a reference.  If you are urgent on merging this pr, it's also ok that I can help to update this later.

Reply:
Thanks, I'd prefer to merge this now, but I'm happy to review a future PR.